src/ToricVarieties/AttributesAndMethods.jl

```diff
@@ -81,10 +81,383 @@ julia> dim(mori)
 1
 ```
 """
-mori_cone( v::NormalToricVariety ) = Cone( v.polymakeNTV.MORI_CONE )
+mori_cone( v::NormalToricVariety ) = Cone( pm_ntv( v ).MORI_CONE )
 export mori_cone
 
 
+@doc Markdown.doc"""
+    affine_open_covering( v::NormalToricVariety )
+
+Computes an affine open cover of the normal toric variety `v`, i.e. returns a list of affine toric varieties.
+
+# Examples
+```jldoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> affine_open_covering( p2 )
+3-element Vector{AffineNormalToricVariety}:
+ A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+ A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+ A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+```
+"""
+function affine_open_covering( v::NormalToricVariety )
+    charts = Vector{AffineNormalToricVariety}(undef, pm_ntv( v ).N_MAXIMAL_CONES)
+    for i in 1:pm_ntv( v ).N_MAXIMAL_CONES
+        charts[ i ] = AffineNormalToricVariety(Cone(Polymake.fan.cone(pm_ntv( v ), i-1)))
+    end
+    return charts
+end
+
+
+@doc Markdown.doc"""
+    affine_open_covering( v::AffineNormalToricVariety )
+
+Computes an affine open cover of the affine normal toric variety `v`, i.e. returns this very variety.
+
+# Examples
+```jdoctest
+julia> C = Oscar.positive_hull([1 0; 0 1])
+A polyhedral cone in ambient dimension 2
+
+julia> antv = AffineNormalToricVariety(C)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> affine_open_covering( antv )
+1-element Vector{AffineNormalToricVariety}:
+ A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+```
+"""
+function affine_open_covering( v::AffineNormalToricVariety )
+    return [ v ]
+end
+export affine_open_covering
+
+
+@doc Markdown.doc"""
+    fan_of_variety( v::NormalToricVariety )
+
+Computes the fan of a normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space( 2 )
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> fan_of_variety( p2 )
+A polyhedral fan in ambient dimension 2
+```
+"""
+function fan_of_variety( v::NormalToricVariety )
+    rays = Matrix{Int}(Oscar.Polymake.common.primitive(pm_ntv( v ).RAYS))
+    cones = v.polymakeNTV.MAXIMAL_CONES
+    return PolyhedralFan( rays, cones )
```

Comment:
```suggestion
   return PolyhedralFan(pm_ntv(v));
```
At some point I had already changed it to the above. Maybe it got removed during migration? Nevertheless, we might already have computed a lot of information about the fan which would be discarded by reconstructing it. Also one would have to deal with lineality, and with the suggestion we don't need to think about those details.

Reply:
I agree. In fact, I would like to commit your suggestion. Still, for some reason it seems to be outdated. I have not updated/pushed since early Monday morning in Europe. Not sure what happened. Can you somehow "refresh"?